cfpb/hubot-cfpb-indexer
# Adding code to create site index and google sheets report

Adding code to create site index and google sheets report. An example of the sheets and index can be viewed here:

https://docs.google.com/spreadsheets/d/11JROb_vrYu2Lr4KD8JegMYBrrMOJcoQjxjbGipokRSw/edit#gid=1345208328

https://raw.githubusercontent.com/cfpb/site-index/5eecdd5fd9bd760d581b5f4bd81b0043bdc38e75/site-index.json

## Changes

- Modified code to create API for indexing the site.

## Testing

1. Setup the correct env variables.
2. Run `.start.sh`
3. Type `cfpbot load index from https://raw.githubusercontent.com/cfpb/site-index/5eecdd5fd9bd760d581b5f4bd81b0043bdc38e75/site-index.json.
4 Type `cfpbot show pages with o-expandable a-btn atomic components`.  It should create a [gist](https://gist.github.com/anonymous/d4e3cadb4ef6370af5b4b16e18b2fa45) url containing a list of pages with those components.
5. Type `cfpbot load index from url https://raw.githubusercontent.com/cfpb/site-index/5eecdd5fd9bd760d581b5f4bd81b0043bdc38e75/site-index.json.
6. The rest of the commands are based on the nightly cron job. 

## Todos
- Write tests.
- Save index on failure so index doesn't have to restart from the beginning.

## Notes
 - Switched from CoffeeScript to JS because I was getting annoyed. 
 - You can run the index locally but you need to switch the default site location
 and GitHub urls.
- Site index time is still too slow. Makes one request every 5 seconds and doesn't index using any query string params.
- Currently not indexing PDFs but that might be a possibility.
